src/Compilers/CSharp/Test/Semantic/Diagnostics/IOperationTests.cs

```diff
@@ -55,5 +55,39 @@ static void Test2(int y, params int[] x)
   IArgument (Matching Parameter: x) (OperationKind.Argument)
     ILiteralExpression (Text: null) (OperationKind.LiteralExpression, Type: null, Constant: null)");
         }
+
+        [Fact]
+        public void DeconstructionAssignmentFromTuple()
+        {
+            var text = @"
+public class C
+{
+    public static void M()
+    {
+        int x, y, z;
+        (x, y, z) = (1, 2, 3);
+        (x, y, z) = new C();
+        var (a, b) = (1, 2);
+    }
+    public void Deconstruct(out int a, out int b, out int c)
+    {
+        a = b = c = 1;
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, parseOptions: TestOptions.RegularWithIOperationFeature);
+            compilation.VerifyDiagnostics();
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+
+            var assignments = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().ToArray();
+            Assert.Equal("(x, y, z) = (1, 2, 3)", assignments[0].ToString());
+            Assert.Equal(null, model.GetOperation(assignments[0]));
```

Comment:
>model.GetOperation(assignments[0]) [](start = 31, length = 34)

I am curious what happens when we get an IOperation node for the containing statement instead? Can we reach to the node for deconstruction by looking at its children. We better not get a node implementing IAssignmentExpression this way.